tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+    if return_code != 0:
+        error_and_cleanup(test_tmp, "Extracting raw data failed (Error {}".format(return_code))
+    print("Comparing known raw data to rsync'd raw data")
+
+    # Construct a list of all files in the Submitty and test versions of reports to make sure the name/number matches
+    known_files = []
+    summary_files = []
+    for f in os.listdir(known_raw_path):
+        known_files.append(f)
+    for f in os.listdir(summary_raw_path):
+        summary_files.append(f)
+
+    if len(known_files) != len(summary_files):
+        file_diff = len(known_files) - len(summary_files)
+        if len(known_files) > len(summary_files):
+            error_and_cleanup(test_tmp,
+                              "There are {} more files in the rsync'd raw_data than expected.".format(file_diff))
+        else:
+            error_and_cleanup(test_tmp,
+                              "There are {} fewer files in the rsync'd raw_data than expected.".format(-1 * file_diff))
+
+    # Verify the content (except for time-dependent fields) of Submitty raw_data files match with test version
+    for f in known_files:
+        contents1 = ""
+        contents2 = ""
+        filename1 = os.path.join(known_raw_path, f)
+        filename2 = os.path.join(summary_raw_path, f)
+
+        try:
+            with open(filename1, 'r') as file1:
+                contents1 = file1.readlines()
+
+            with open(filename2, 'r') as file2:
+                contents2 = file2.readlines()
+        except Exception as e:
+            error_and_cleanup(test_tmp, "{}".format(e))
+
+        # Use filters to avoid time-dependent fields and speed up comparison
+        filter1 = filter(remove_extra_raw_data_fields, contents1)
+        filter2 = filter(remove_extra_raw_data_fields, contents2)
+        same_flag = True
+        for x, y in zip(filter1, filter2):
+            if x != y:
+                same_flag = False
+        if not same_flag:
+            error_and_cleanup(test_tmp, "{} and {} differ".format(filename1, filename2))
+
+    print("All raw files match")
+
+    # PRECONDITION: Input at this point is verified. Test running Rainbow Grades.
+    print("Running Rainbow Grades on rsync'd data. This could take several minutes.")
+    make_output = ""
+    try:
+        make_output = subprocess.check_output(["make"])
+    except subprocess.CalledProcessError as e:
+        error_and_cleanup(test_tmp, "Make failed with code {}".format(e.returncode))
+
+    if not os.path.exists(os.path.join(summary_tmp, "output.html")):
+        error_and_cleanup(test_tmp, "Failed to create output.html")
+
+    print("output.html generated")
+
+    print("Checking summary files against expected summaries.")
+    # Verify that a valid copy of output.html was sent to all_students_summary_html
+    make_output = make_output.splitlines()
+    make_output = make_output[-1].strip()  # Get the RUN COMMAND LINE
+    make_output = make_output.split(b'/')
```

Comment:
I'd probably just do `make_output.decode("utf-8")` on the subprocess code instead of messing around with binary strings.